packages/salesforcedx-vscode-lwc/test/vscode-integration/intellisense.test.ts

```diff
@@ -8,217 +8,264 @@
+    let endPosition: Position;
+    let selection: Selection;
 
-  afterEach(async () => {
-    try {
-      await vscode.commands.executeCommand(
-        'workbench.action.closeActiveEditor'
+    beforeEach(async() => {
+      lwcDir = path.join(
+        vscode.workspace.workspaceFolders![0].uri.fsPath,
+        'force-app',
+        'main',
+        'default',
+        'lwc'
       );
-    } catch (e) {
-      throw e;
-    }
-  });
+      lwcExtension = extensions.getExtension(
+        'salesforce.salesforcedx-vscode-lwc'
+      ) as Extension<any>;
+      await lwcExtension.activate();
+      docUri = Uri.file(path.join(lwcDir, 'hello', 'hello.js'));
+      doc = await workspace.openTextDocument(docUri);
+      editor = await window.showTextDocument(doc);
+      startPosition = new Position(1, 0);
+      // To provide valid arguments to selection in afterEach, setting default value of endPosition same as startPosition. 
+      endPosition = new Position(1,0);
+    });
 
-  it('lwc extension activation', async function() {
-    expect(lwcExtension.isActive);
-  }).timeout(10000);
-
-  /**
-   * Test that lwc markup intellisense includes standard lwc tags and custom lwc tags
-   */
-  it('LWC Markup Intellisense', async () => {
-    const docUri = vscode.Uri.file(path.join(lwcDir, 'hello', 'hello.html'));
-    const doc = await vscode.workspace.openTextDocument(docUri);
-    const editor = await vscode.window.showTextDocument(doc);
-
-    // We have to have some text or we'll just get generic completions
-    const text = '<c-';
-    const startPosition = new vscode.Position(5, 0);
-    const endPosition = new vscode.Position(
-      startPosition.line,
-      startPosition.character + text.length
-    );
-    const rangeReplace = new vscode.Range(startPosition, endPosition);
-    await editor.edit(editBuilder => {
-      editBuilder.replace(rangeReplace, text);
+    afterEach(() => {
+      try {
+        selection = new Selection(startPosition, endPosition);
+        // We need to clear the editor's line or the input text will change and test will fail.
+        editor.edit(editBuilder => {
+          editBuilder.delete(selection)
+        });
+      } catch (e) {
+        throw e;
+      }
     });
 
-    try {
-      // NOTE: Because the completion providers always returns all possible results and then VSCode
-      // does the filtering based on what is typed, we have no good way of testing what vscode is
-      // actually displaying to the user based on what we typed
-      // TODO - investigate why this only happens on markup
-      await testCompletion(docUri, endPosition, {
-        items: [
-          {
-            label: 'lightning-accordion',
-            kind: vscode.CompletionItemKind.Property
-          },
-          { label: 'c-hello-binding', kind: vscode.CompletionItemKind.Property }
-        ]
+    it('LWC Extension Activation', async () => {
+      expect(lwcExtension.isActive);
+    });
```

Comment:
This test can go at the top level describe. Esp if we move the activate wait to the top level before.